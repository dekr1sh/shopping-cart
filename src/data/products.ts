export interface Product {
  id: number
  name: string
  brand: string
  price: number
  image: string
  description: string
}

export const products: Product[] = [
  {
    id: 1,
    name: "Air Max 90",
    brand: "Nike",
    price: 9999.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-1.webp",
    description:
      "The Nike Air Max 90 stays true to its OG running roots with the iconic Waffle outsole, stitched overlays and classic TPU details. Classic colors celebrate your fresh look while Max Air cushioning adds comfort to your journey. This legendary shoe continues to deliver the same lightweight cushioning that made it famous.",
  },
  {
    id: 2,
    name: "NMD R1",
    brand: "Adidas",
    price: 10999.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-2.webp",
    description:
      "The NMD R1 combines the best of adidas' technologies and design to create a shoe that's ready for whatever the streets throw at you. The Boost midsole provides responsive cushioning, while the Primeknit upper offers a sock-like fit. The signature EVA midsole plugs add a distinctive look that's instantly recognizable.",
  },
  {
    id: 3,
    name: "990v5",
    brand: "New Balance",
    price: 13999.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-3.webp",
    description:
      "The 990's original designers were tasked with creating the single best running shoe on the market. The finished product more than delivered. When it hit shelves for the first time in 1982 the 990 sported an elegantly understated grey colorway, and a then unheard of three-figure price tag. For avid runners and ahead of the curve tastemakers alike, the 990 was a mark of quality and superior taste.",
  },
  {
    id: 4,
    name: "Ultraboost 21",
    brand: "Adidas",
    price: 11999.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-4.webp",
    description:
      "Ultraboost 21. The new era in adidas running technology. Incredible energy return, responsive comfort and weight-defying strength for runners and non-runners alike. The Ultraboost 21 features a BOOST midsole for incredible energy return, a Primeknit+ upper for adaptive support and comfort, and a Linear Energy Push system for increased responsiveness.",
  },
  {
    id: 5,
    name: "Air Force 1",
    brand: "Nike",
    price: 7999.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-5.webp",
    description:
      "The radiance lives on in the Nike Air Force 1, the basketball original that puts a fresh spin on what you know best: crisp leather, bold colors and the perfect amount of flash to make you shine. The low-cut silhouette creates a clean, streamlined look while the padded collar feels soft and comfortable.",
  },
  {
    id: 6,
    name: "RS-X³",
    brand: "Puma",
    price: 8499.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-6.webp",
    description:
      "The RS-X³ takes the RS-X to a new level with a design that's bigger, bolder and better. The shoe features a bulky silhouette with a mesh and textile upper, a padded collar and tongue, and a cushioned midsole for maximum comfort. The RS (Running System) technology provides cushioning and support for all-day wear.",
  },
  {
    id: 7,
    name: "Suede Classic",
    brand: "Puma",
    price: 5499.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-7.webp",
    description:
      "The Suede has been a street icon since its introduction in 1968. It's been worn by icons of every generation, and it's stayed classic through it all. This version of the Suede features a premium suede upper, a padded collar and tongue, and the iconic formstrip design. The rubber outsole provides traction and durability for everyday wear.",
  },
  {
    id: 8,
    name: "574 Core",
    brand: "New Balance",
    price: 6499.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-8.webp",
    description:
      "The 574 Core features a premium upper with suede and mesh materials for the perfect combination of style and comfort. The ENCAP midsole technology provides support and maximum durability, while the EVA foam midsole delivers cushioning. The rubber outsole offers traction and durability for all-day wear.",
  },
  {
    id: 9,
    name: "Dunk Low",
    brand: "Nike",
    price: 8999.99,
    image: "/home/de_kr1sh/web-development/projects/shopping-cart/src/assets/product-images/sneaker-9.webp",
    description:
      "Created for the hardwood but taken to the streets, the Nike Dunk Low is your emblem of individuality. The university-inspired design channels '80s vibes with bold team colors and a padded, low-cut collar. The padded, low-cut collar lets you take your game anywhere—in comfort.",
  },
]