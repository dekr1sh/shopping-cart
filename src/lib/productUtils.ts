import type { Product } from "@/data/products";

export function getFeaturedProducts(products: Product[], count = 3) {
  return products.slice(0, count);
}

export function getBrands(products: Product[]) {
  const brands = new Set(products.map((product) => product.brand));
  return Array.from(brands);
}

export function getProductsByBrand(products: Product[], brandNames: string[]) {
  if (!brandNames.length) return products;
  return products.filter((product) => brandNames.includes(product.brand));
}

export function getProductsByPriceRange(products: Product[], min?: number, max?: number) {
  if(min === undefined && max === undefined) return products;

  return products.filter((product) => {
    const aboveMin = min !== undefined ? product.price >= min : true;
    const belowMax = max !== undefined ? product.price <= max : true;
    return aboveMin && belowMax;
  });
}

export function getFilteredProducts(products: Product[], brandNames: string[], min?: number, max?: number) {
  let filteredProducts = products;

  filteredProducts = getProductsByBrand(filteredProducts, brandNames);

  filteredProducts = getProductsByPriceRange(filteredProducts, min, max);

  return filteredProducts;
}