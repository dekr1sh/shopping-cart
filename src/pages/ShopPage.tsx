import { useState} from "react"
import { Button } from "@/components/ui/button"
import { Card, CardFooter, CardHeader, CardTitle } from "@/components/ui/card"
import { Link } from "react-router-dom"
import { products } from "@/data/products"
import { getFilteredProducts } from "@/lib/productUtils"
import FilterSection from "@/components/FilterSection"
import placeholder from "@/assets/other-images/placeholder.png"

export default function ShopPage() {
  const [selectedBrands, setSelectedBrands] = useState<string[]>([])
  const [minPrice, setMinPrice] = useState<number | undefined>(undefined)
  const [maxPrice, setMaxPrice] = useState<number | undefined>(undefined)

  const filteredProducts = getFilteredProducts(products, selectedBrands, minPrice, maxPrice);

  const handleBrandChange = (brand: string, checked: boolean) => {
    if (checked) {
      setSelectedBrands((prev) => [...prev, brand])
    } else {
      setSelectedBrands((prev) => prev.filter((b) => b !== brand))
    }
  }

  const clearFilters = () => {
    setSelectedBrands([])
    setMinPrice(undefined)
    setMaxPrice(undefined)
  }

  return (
    <div className="px-4 py-8">
      <h1 className="text-3xl font-bold mb-8">Shop All Sneakers</h1>

      <div className="flex flex-col md:flex-row gap-6 mb-8">
        <div className="w-full md:w-1/4 space-y-4">
          <FilterSection
            selectedBrands={selectedBrands}
            minPrice={minPrice}
            maxPrice={maxPrice}
            onBrandChange={handleBrandChange}
            onMinPriceChange={setMinPrice}
            onMaxPriceChange={setMaxPrice}
            onClearFilters={clearFilters}
          />
        </div>

        <div className="w-full md:w-3/4">
          {filteredProducts.length === 0 ? (
            <div className="text-center py-12">
              <h3 className="text-lg font-medium mb-2">No products found</h3>
              <p className="text-muted-foreground mb-4">Try adjusting your filters to find what you're looking for.</p>
              <Button onClick={clearFilters}>Clear Filters</Button>
            </div>
          ) : (
            <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6">
              {filteredProducts.map((product) => (
                <Link to={`/product/${product.id}`} key={product.id}>
                  <Card className="overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer">
                    <div className="relative h-48 bg-muted">
                      <img
                        src={product.image || placeholder}
                        alt={product.name}
                        className="object-cover w-full h-full"
                      />
                    </div>
                    <CardHeader className="p-4 pb-0">
                      <div className="flex justify-between items-start">
                        <CardTitle className="text-lg">{product.name}</CardTitle>
                        <span className="font-bold">â‚¹{product.price.toFixed(2)}</span>
                      </div>
                      <p className="text-sm text-muted-foreground">{product.brand}</p>
                    </CardHeader>
                    <CardFooter className="p-4 pt-2">
                      <Button variant="secondary" className="w-full">
                        View Details
                      </Button>
                    </CardFooter>
                  </Card>
                </Link>
              ))}
            </div>
          )}
        </div>
      </div>
    </div>
  )
}
