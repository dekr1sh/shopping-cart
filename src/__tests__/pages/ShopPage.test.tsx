import { describe, it, expect, vi } from "vitest"
import { screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { render } from "@/__tests__/test-utils"
import ShopPage from "@/pages/ShopPage"
import * as productUtils from "@/lib/productUtils"
import { products } from "@/data/products"

vi.mock("@/components/FilterSection", () => ({
  default: vi.fn(({ onBrandChange, onClearFilters }) => (
    <div>
      <label>
        <input
          type="checkbox"
          name="Nike"
          onChange={(e) => onBrandChange("Nike", e.target.checked)}
        />
        Nike
      </label>
      <button onClick={onClearFilters}>Clear Filters</button>
    </div>
  ))
}))

describe("ShopPage", () => {
  it("renders correctly when there are no filtered products", () => {
    vi.spyOn(productUtils, "getFilteredProducts").mockReturnValue([])

    const { container } = render(<ShopPage />)
    expect(container).toMatchSnapshot()
  })

  it("renders correctly when there are filtered products", () => {
    vi.spyOn(productUtils, "getFilteredProducts").mockReturnValue(products)

    const { container } = render(<ShopPage />)
    expect(container).toMatchSnapshot()
  })

  it("filters products by brand when checkbox is selected", async () => {
    const getFilteredProductsSpy = vi.spyOn(productUtils, "getFilteredProducts")

    render(<ShopPage />)

    const nikeCheckbox = screen.getByLabelText("Nike")
    await userEvent.click(nikeCheckbox)

    expect(getFilteredProductsSpy).toHaveBeenLastCalledWith(products, ["Nike"], undefined, undefined);

    await userEvent.click(nikeCheckbox);
    expect(getFilteredProductsSpy).toHaveBeenLastCalledWith(products, [], undefined, undefined);
  })

  it("clears filters when Clear Filters button is clicked", async () => {
    const getFilteredProductsSpy = vi.spyOn(productUtils, "getFilteredProducts")

    render(<ShopPage />)

    const nikeCheckbox = screen.getByLabelText("Nike")
    await userEvent.click(nikeCheckbox)

    expect(getFilteredProductsSpy).toHaveBeenLastCalledWith(products, ["Nike"], undefined, undefined);

    await userEvent.click(screen.getByText("Clear Filters"))
    expect(getFilteredProductsSpy).toHaveBeenLastCalledWith(products, [], undefined, undefined);
  })
})
