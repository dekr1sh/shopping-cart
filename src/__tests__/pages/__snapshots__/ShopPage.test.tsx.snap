// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShopPage > renders correctly when there are filtered products 1`] = `
<div>
  <div
    class="px-4 py-8"
  >
    <h1
      class="text-3xl font-bold mb-8"
    >
      Shop All Sneakers
    </h1>
    <div
      class="flex flex-col md:flex-row gap-6 mb-8"
    >
      <div
        class="w-full md:w-1/4 space-y-4"
      >
        <div>
          <label>
            <input
              name="Nike"
              type="checkbox"
            />
            Nike
          </label>
          <button>
            Clear Filters
          </button>
        </div>
      </div>
      <div
        class="w-full md:w-3/4"
      >
        <div
          class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6"
        >
          <a
            data-discover="true"
            href="/products/1"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="Air Max 90"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    Air Max 90
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    9999.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Nike
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/2"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="NMD R1"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    NMD R1
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    10999.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Adidas
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/3"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="990v5"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    990v5
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    13999.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  New Balance
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/4"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="Ultraboost 21"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    Ultraboost 21
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    11999.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Adidas
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/5"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="Air Force 1"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    Air Force 1
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    7999.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Nike
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/6"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="RS-X³"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    RS-X³
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    8499.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Puma
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/7"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="Suede Classic"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    Suede Classic
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    5499.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Puma
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/8"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="574 Core"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    574 Core
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    6499.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  New Balance
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
          <a
            data-discover="true"
            href="/products/9"
          >
            <div
              class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm overflow-hidden h-full transition-transform hover:scale-105 cursor-pointer"
              data-slot="card"
            >
              <div
                class="relative h-48 bg-muted"
              >
                <img
                  alt="Dunk Low"
                  class="object-cover w-full h-full"
                  src="/src/assets/other-images/placeholder.png"
                />
              </div>
              <div
                class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 p-4 pb-0"
                data-slot="card-header"
              >
                <div
                  class="flex justify-between items-start"
                >
                  <div
                    class="font-semibold text-lg"
                    data-slot="card-title"
                  >
                    Dunk Low
                  </div>
                  <span
                    class="font-bold"
                  >
                    ₹
                    8999.99
                  </span>
                </div>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Nike
                </p>
              </div>
              <div
                class="flex items-center [.border-t]:pt-6 p-4 pt-2"
                data-slot="card-footer"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
                  data-slot="button"
                >
                  View Details
                </button>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
     
</div>
`;

exports[`ShopPage > renders correctly when there are no filtered products 1`] = `
<div>
  <div
    class="px-4 py-8"
  >
    <h1
      class="text-3xl font-bold mb-8"
    >
      Shop All Sneakers
    </h1>
    <div
      class="flex flex-col md:flex-row gap-6 mb-8"
    >
      <div
        class="w-full md:w-1/4 space-y-4"
      >
        <div>
          <label>
            <input
              name="Nike"
              type="checkbox"
            />
            Nike
          </label>
          <button>
            Clear Filters
          </button>
        </div>
      </div>
      <div
        class="w-full md:w-3/4"
      >
        <div
          class="text-center py-12"
        >
          <h3
            class="text-lg font-medium mb-2"
          >
            No products found
          </h3>
          <p
            class="text-muted-foreground mb-4"
          >
            Try adjusting your filters to find what you're looking for.
          </p>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 cursor-pointer"
            data-slot="button"
          >
            Clear Filters
          </button>
        </div>
      </div>
    </div>
  </div>
     
</div>
`;
